### PR TITLE
doc/kconfig: add a note about possible CPU intermediate hierarchy level

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

As requested in https://github.com/RIOT-OS/RIOT/pull/14998#issuecomment-700617037, this PR slightly extends the Kconfig documentation page with a mention saying it's allowed to add intermediate grouping levels in CPU for convenience if it better reflects the hierarchy used by a given vendor.

Better wording is welcome !

### Testing procedure

Just read!

### Issues/PRs references

Documentation about this was requested in https://github.com/RIOT-OS/RIOT/pull/14998#issuecomment-700617037
Based on #14998

### CODE REVIEW

**Comment by fjmolinas** [comment]
Needs a rebase now.

**Comment by aabadie** [comment]
rebased

File: doc/doxygen/src/kconfig/kconfig.md [hunk 1]
```diff
@@ -352,7 +352,9 @@ Features may be provided by any hierarchy symbol. Usually symbols are selected
 from more specific to less specific. This means that a `CPU_MODEL_<model>`
 symbol usually would select the correspondent `CPU_FAM_<family>` symbol,
 which would in turn select the `CPU_CORE_<core>`. This may change in some cases
-where `CPU_COMMON_` symbols are defined to avoid repetition.
+where `CPU_COMMON_` symbols are defined to avoid repetition. For convenience and
+if it makes sense within a CPU vendor family, it's also allowed to use
+intermediate grouping levels, like the `CPU_LINES_<xxx>` used for STM32.
```
Comment (leandrolanzieri) [inline doc/doxygen/src/kconfig/kconfig.md:357]:
```suggestion
intermediate grouping levels, like `CPU_LINE_<xxx>` used for STM32.
```

Comment (aabadie) [reply doc/doxygen/src/kconfig/kconfig.md]: Thanks! Fixed and directly squashed.

**Review by leandrolanzieri — APPROVED** [review]
Reflects what was discussed in #14998. ACK!

**Comment by aabadie** [comment]
Thanks for the review!

**Pull request merged**